src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -3117,6 +3290,337 @@ NetworkOPsImp::unsubAccountInternal(
     }
 }
 
+void
+NetworkOPsImp::addAccountHistoryJob(SubAccountHistoryInfoWeak subInfo)
+{
+    app_.getJobQueue().addJob(
+        jtCLIENT, "AccountHistoryTxStream", [this, subInfo](Job&) {
+            auto const& accountId = subInfo.index_->accountId_;
+            auto& lastLedgerSeq = subInfo.index_->historyLastLedgerSeq_;
+            auto& txHistoryIndex = subInfo.index_->historyTxIndex_;
+
+            JLOG(m_journal.trace())
+                << "AccountHistory job for account " << toBase58(accountId)
+                << " started. lastLedgerSeq=" << lastLedgerSeq;
+
+            auto isFirstTx = [&](std::shared_ptr<Transaction> const& tx,
+                                 std::shared_ptr<TxMeta> const& meta) -> bool {
+                /*
+                 * genesis account: first tx is the one with seq 1
+                 * other account: first tx is the one created the account
+                 */
+                if (accountId == genesisAccountId)
+                {
+                    auto stx = tx->getSTransaction();
+                    if (stx->getAccountID(sfAccount) == accountId &&
+                        stx->getSeqProxy().value() == 1)
+                        return true;
+                }
+
+                for (auto& node : meta->getNodes())
+                {
+                    if (node.getFieldU16(sfLedgerEntryType) != ltACCOUNT_ROOT)
+                        continue;
+
+                    if (node.isFieldPresent(sfNewFields))
+                    {
+                        if (auto inner = dynamic_cast<const STObject*>(
+                                node.peekAtPField(sfNewFields));
+                            inner)
+                        {
+                            if (inner->isFieldPresent(sfAccount) &&
+                                inner->getAccountID(sfAccount) == accountId)
+                            {
+                                return true;
+                            }
+                        }
+                    }
+                }
+
+                return false;
+            };
+
+            auto send = [&](Json::Value const& jvObj,
+                            bool unsubscribe) -> bool {
+                if (auto sptr = subInfo.sinkWptr_.lock(); sptr)
+                {
+                    sptr->send(jvObj, true);
+                    if (unsubscribe)
+                        unsubAccountHistory(sptr, accountId, false);
+                    return true;
+                }
+
+                return false;
+            };
+
+            auto db = dynamic_cast<RelationalDBInterfaceSqlite*>(
+                &app_.getRelationalDBInterface());
+            if (!db)
+            {
+                JLOG(m_journal.debug())
+                    << "AccountHistory job for account " << toBase58(accountId)
+                    << " DB interface type wrong";
+                send(rpcError(rpcINTERNAL), true);
+                return;
+            }
```

Comment:
We need to handle this case, and not return an error. Reporting mode can do what you want here. The Relational DB Interface was just poorly designed and thus you need to call a different function in this case.

Reply:
Thanks for pointing out the reporting mode. I read the code, `RelationalDBInterfacePostgres` uses different functions (different interface) as you said, and the guarantee of the interface is not that clear to me from the docs. Of course it seems doable by following the example of `accountTx` in reporting mode. And more testing will be needed. 

But before making the change, I am wondering if it is needed, because as I understand, we don't plan to add this to reporting mode so we won't have a `RelationalDBInterfacePostgres` database backend. 

Reply:
We have to add this to reporting mode, because our public client handlers on testnet and mainnet will be running in reporting mode. It will make it harder for people to use this if we don't add it to reporting mode.

Reply:
Right, added